Sources/Tracking/Service/Request/ScreenViewDefaultData.swift

```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+// ScreenViewData wraps an encoder to allow returning of custom data for automatic screenview tracking
+public struct ScreenViewData: Encodable {
```

Comment:
Is this for doing custom attributes? 

Wondering what the need for this struct is since `trackEvent()` doesn't use something like this?

Reply:
yeah I was fighting with some interface things. came up with an alternative

Reply:
that alternative is `[String:AnyEncodable]` and another protocol signature extension. Do you think we also need to support `[String:Any]` or are people likely to be able to conform to `[String:AnyEncodable]` (calling back to the convo yesterday with Lasting, trying to see whether this'll help them as well)

Reply:
I looked at what other SDKs use as a type signature and `[String: Any]` is preferred. [Here is an idea](https://github.com/segmentio/analytics-swift/blob/c5a7f4ffc2635fd834ae029f130744ca87c30a94/Sources/Segment/Events.swift#L139-L149) for how to get started on that. 

To get `[String: Any]` working with the SDK, it might require more work since `Codable` may or may not be compatible with the `JSON` data type. Should we ignore `[String: Any]` in this PR and move that into a new PR where we make *all* custom attributes in the SDK compatible with `[String: Any]`?